pkg/operator/bootstrapteardown/waitforceo.go

```diff
@@ -73,3 +83,114 @@ func done(etcd *operatorv1.Etcd) (bool, error) {
 	klog.Infof("Still waiting for the cluster-etcd-operator to bootstrap")
 	return false, nil
 }
+
+func WaitForKubeApiServerRollout(ctx context.Context, config *rest.Config) error {
+	operatorClient, err := operatorversionedclient.NewForConfig(config)
+	if err != nil {
+		klog.Errorf("WaitForKubeApiServerRollout: error getting operator client: %#v", err)
```

Comment:
General structure comment.  I recommend rolling these up.  You can use a `%w` now which will help. Then have a higher level emit an event with the error details.  It will also appear in your log, but events will come back from insights.